### ISSUE TITLE
Trigger service returns inconsistent results

### ISSUE THREAD

**Issue by gaborh-da** [digital-asset/daml#8703]
After starting a couple of triggers with the trigger service, I wanted to check one of them:
```
~/work/ex-know-your-customer$ curl 127.0.0.1:8088/v1/triggers/0bf72a80-abc5-48db-b52d-6e71593caf2f
{"result":{"party":"Operator","status":"running","triggerId":"ea8afc633f41996ca9d4781419e17879b6d8258f7f2466b23cccf55041913b4e:DA.RefApps.KnowYourCustomer.Triggers.TimeUpdater:timeUpdaterTrigger"},"status":200}
```

The same trigger *is not shown running* when I check per party. (same endpoints, Operator)
```
~/work/ex-know-your-customer$ curl 127.0.0.1:8088/v1/triggers?party=Operator
{"result":{"triggerIds":[]},"status":200}
```

Reproduction:
- use this draft https://github.com/digital-asset/ex-know-your-customer/pull/32
- follow the README (one needs to start Sandbox, execute `scripts/startTriggerService.sh` and then check)

**Comment by aherrmann-da** [comment]
I've tried to reproduce this, but wasn't able to so far. Here are the steps I've performed:
- Check-out https://github.com/digital-asset/ex-know-your-customer/pull/32/commits/c028180c37c09ca10f3f79d87785c993af82d04d
- Fix the `startTriggerService.sh` script as shown in https://github.com/digital-asset/ex-know-your-customer/pull/32#discussion_r567682384.
- Start `daml start --sandbox-option --address=localhost --sandbox-option -w` in one terminal and wait for it to be ready.
- Start `./scripts/startTriggerService.sh localhost 6865 .daml/dist/know-your-customer-1.0.0.dar` in another terminal and wait for it to be ready.
- Look for the `triggerId` of the `DA.RefApps.KnowYourCustomer.Triggers.TimeUpdater:timeUpdaterTrigger` trigger runner in the output of `startTriggerService.sh`.
- Compare the outputs of
  ```
  $ curl 127.0.0.1:8088/v1/triggers/$TRIGGER_ID
  $ curl 127.0.0.1:8088/v1/triggers?party=Operator
  ```

I haven't observed the described inconsistency within three trials.
I've tested on Ubuntu 20.10 with DAML SDK 1.8.0 (as specified in `daml.yaml`).

@gaborh-da Are there any steps I'm missing or any potentially relevant differences in the test environment, e.g. daml version?
